### ISSUE TITLE
'sizes' in TileGrid is ignored

### ISSUE THREAD

**Issue by mst2020** [openlayers/openlayers#11522]
Hello,

and thanks for your time and effort to understand my problem description. :)

Greetings.

**Describe the bug**
Maybe i'm too dumb to understand it, but when i set the 'sizes' property of the [TileGrid](https://openlayers.org/en/latest/apidoc/module-ol_tilegrid_TileGrid-TileGrid.html), the [Tile Image Source](https://openlayers.org/en/latest/apidoc/module-ol_source_TileImage.html) is still making requests to tiles outside of min/max x/y.

This is my grid as [TileDebug](https://openlayers.org/en/latest/apidoc/module-ol_source_TileDebug-TileDebug.html) shows it. I restrict it with an extent, for sure, but also with "sizes" as you can see below the image:

![ol](https://user-images.githubusercontent.com/10521032/92332617-90425300-f07f-11ea-91a7-2ec8bae30f69.jpg)

The TileGrid is successfully  creating the "[fullTileRanges_](https://github.com/openlayers/openlayers/blob/4ddf5c0c3eaebc5f5798500cb8ece533e70dda20/src/ol/tilegrid/TileGrid.js#L183)"
```
fullTileRanges_: Array(4)
0: t {minX: 0, maxX: 0, minY: 0, maxY: 1}
1: t {minX: 0, maxX: 1, minY: 0, maxY: 3}
2: t {minX: 0, maxX: 1, minY: 0, maxY: 3}
3: t {minX: 0, maxX: 10, minY: 0, maxY: 15}
```

**1: t {minX: 0, maxX: 1, minY: 0, maxY: 3}**

Until this point everything seems fine, but "fullTileRanges_" is ignored by the Layer/Source and creates requests for the following coords (z, x, y):
```
OutOfRange [1, 2, 0]
OutOfRange [1, 2, 1]
OutOfRange [1, 2, 2]
OutOfRange [1, 2, 3]
```
_(I catch those requests in my "tileLoadFunction" (and tileloadstart event))_

Hm ... maybe its a hint that its only exceeding the X bounds ... ?

**To Reproduce**
Define your own TileGrid (restricted by "sizes" property) for your TileImage for your layer.Tile ...

**Expected behavior**
TileImage should *not* query tiles which are outside of the bounds of sizes/fullTileRanges_.

**Comment by ahocevar** [comment]
Did you configure your source with `wrapX: false`? This issue would be easier to evaluate if you'd share your code.

**Comment by mst2020** [comment]
> Did you configure your source with `wrapX: false`? This issue would be easier to evaluate if you'd share your code.

I'll tried it with "wrapX" but with no success.

i build an jsfiddle and fortunately for us the error happens there again :)
https://jsfiddle.net/q65v8khz

**Comment by ahocevar** [comment]
Ok, this is because you configured the tile grid with an `extent` in addition to `sizes`. You'd need to remove `extent` from the configuration, and configure the extent's top-left corner as `origin` instead.

Maybe the documentation for `sizes` could be improved here a bit. Currently it says

> A grid extent can be configured in addition, and will further limit the extent for which tile requests are made by sources.

It should probably say something like

> The `sizes` option will be ignored when an `extent` is configured.

**Comment by mst2020** [comment]
You are my hero! :) Many thanks, its working now.

> The `sizes` option will be ignored when an `extent` is configured.

That could have avoid this issue. ;)

Have a nice week!

greetings

**Comment by mike-000** [comment]
This also seems to apply to WMTSTileGrid and is resulting in 404 errors in the Print to scale example.
As well as a further documentation update `WMTSTileGrid#createFromCapabilitiesMatrixSet` should probably ensure that only one of sizes and extent are set in the constructor (with valid sizes having precedence).

**Comment by ahocevar** [comment]
@mike-000 Would you be able to create a pull request fixing the WMTS tile grid issues you mentioned?

**Comment by ahocevar** [comment]
@mike-000 Actually looking closer at this, I see there is an issue with extent vs. sizes that goes deeper. I'm currently reworking #11532, and as a result the example you mentioned should work afterwards. So nothing to do at your end at the moment.

**Comment by ahocevar** [comment]
#11532 now fixes this issue - not by changing the documentation, but by implementing what the documentation says.

**Comment by ahocevar** [comment]
Fixed with #11532
